dart/dynamics/BodyNode.cpp

```diff
@@ -1339,6 +1339,14 @@ void BodyNode::notifyTransformUpdate()
 {
   notifyVelocityUpdate(); // Global Velocity depends on the Global Transform
 
+  // Jacobian calculations are dependent on the parent's world transform, but
+  // not on the world transform of their own BodyNode, so they must be dirtied
+  // regardless of whether the world transform of this BodyNode is already dirty
```

Comment:
I agree with that the transform of a parent BodyNode can be changed without its child BodyNodes being informed that their Jacobians are now dirty. Suppose we have a serial chain and the position of i-th joint changed then only the child BodyNodes Jacobian will be dirty when all the Jacobians of descendant BodyNodes should be dirty as well. In that sense, it seems good solution to set the flags at here instead of Joint::notifyPositionUpdate().

One concern is that we might want to update this comment a bit. If I'm correct, the Jacobian of a BodyNode is actually dependent on the parent's Jacobian and the relative transform (or the joint positions of the parent Joint) of the BodyNode (to transform the coordinates of the parent's Jacobian from the parent's to the BodyNode's). 

Also, it would be good to clarify more why and when these flags must be dirtied regardless of whether the relative transform of this BodyNode is already dirty. Basically, we need this for the case, when BodyNode::notifyTransformUpdate() is called, Jacobian is updated while the relative transform of this BodyNode is not updated yet. However, this case wouldn't happen if the BodyNode has a parent BodyNode. The relative transform will be consequently updated when the Jacobian is updated to be used in transforming the coordinates of the parent's Jacobian. You can check this in BodyNode::updateBodyJacobian(). That being said, we place blew code here (before the relative transform's dirtiness check) for the case this BodyNode doesn't have parent BodyNode. Maybe this might be too much detailed note but it would prevent us from spending time to interpret again the intention of this code in the future.